### DISCUSSION TITLE
Failed to create a tunnel to Apple TV 4G with iOS 17.3

### DISCUSSION THREAD

**Discussion by fjanisze** [doronz88/pymobiledevice3#877] [Q&A]
While executing the command: `sudo python3 -m pymobiledevice3 remote start-tunnel` the script fails with:

```
  File "/<CUT>/pymobiledevice3/remote/core_device_tunnel_service.py", line 473, in _request_pair_consent
    return PairConsentResult(public_key=data[PairingDataComponentType.PUBLIC_KEY],
KeyError: EnumIntegerString.new(3, 'PUBLIC_KEY')
```

I modified the script code (`core_device_tunnel_service.py`) to print the result of `data`:

```
        data = self.decode_tlv(PairingDataComponentTLVBuf.parse(
            pairingData))
        print(data)
        return PairConsentResult(public_key=data[PairingDataComponentType.PUBLIC_KEY],
                                 salt=data[PairingDataComponentType.SALT])
```

and it contains:

`{EnumIntegerString.new(7, 'ERROR'): b'\x03', EnumIntegerString.new(8, 'RETRY_DELAY'): b"\x08'", EnumIntegerString.new(6, 'STATE'): b'\x02'}`

while `pairingData` contains:

`{'pairingData': {'_0': {'kind': 'setupManualPairing', 'data': b'\x07\x01\x03\x08\x02| \x06\x01\x02', 'startNewSession': False}}}
`
What could be the issue?

**Comment by doronz88** [comment]
Maybe you clicked the don't trust option? Try to remove paired device history
